apps/openmw/mwphysics/physicssystem.cpp

```diff
@@ -1246,11 +1254,42 @@ namespace MWPhysics
             mStandingCollisions.clear();
         }
 
+        if (mNumThreads > 1)
+        {
+            int currentThread = 0;
+            std::vector<std::thread> threads(mNumThreads);
+            for(auto&& thread : threads)
+            {
+                thread = std::thread(&PhysicsSystem::solveTask, this, currentThread, mNumThreads, numSteps);
+                currentThread++;
+            }
+
+            for(auto&& thread : threads)
+            {
+                thread.join();
+            }
+        }
+        else
+            solveTask(0, 1, numSteps);
+
+
+        mMovementQueue.clear();
+
+        return mMovementResults;
+    }
+
+    void PhysicsSystem::solveTask(int offset, int module, int numSteps)
+    {
+        int i = 0;
         const MWWorld::Ptr player = MWMechanics::getPlayer();
         const MWBase::World *world = MWBase::Environment::get().getWorld();
         PtrVelocityList::iterator iter = mMovementQueue.begin();
         for(;iter != mMovementQueue.end();++iter)
```

Comment:
You protect `mMovementResults` with mutex, but not protect `mMovementQueue` though they have the same type, so are not thread-safe. Or this vector can not change during physics update? If so, I think it is better to explicit comment that here (or even better, as doxygen comment on `mMovementQueue`)